src/viam/trajex/totg/trajectory.cpp

```diff
@@ -348,10 +350,17 @@ enum class integration_event : std::uint8_t {
                         const auto [s_dot_max_acc, s_dot_max_vel] =
                             compute_velocity_limits(q_prime, q_double_prime, opt.max_velocity, opt.max_acceleration, opt.epsilon);
 
+                        // NOTE: `Divergent Behavior 2`: If this switching point is infeasible with respect to the
+                        // velocity limit curve, reject it.
+                        if (opt.epsilon.wrap(s_dot_max_vel) < opt.epsilon.wrap(s_dot_max_acc)) {
+                            first_extremum.reset();
+                            return;
+                        }
+
                         // Validate: acceleration limit curve must have local minimum (derivative changes negative to positive)
                         // Sample slightly before and after the extremum, clamped to segment bounds
                         //
-                        // TODO: Replace numerical derivative check with analytical derivative computation.
+                        // TODO(RSDK-12850): Replace numerical derivative check with analytical derivative computation.
```

Comment:
I think this is the right ticket? But it is also mentioned in the negative step case below, so maybe I'm conflating things, and this should in fact be a different ticket.

Reply:
I think so as well, the Epic does not have tickets other than this relating to numerical derivatives.